src/grid/GridLayout.tsx

```diff
@@ -41,135 +41,144 @@ export const makeGridLayout: CallLayout<GridLayoutModel> = ({
-          // width (w = (W - (c - 1) * g) / c) and solve for c, we get
-          // c = (W + g) / (w + g).
-          columns = Math.floor((width + gap) / (slotMinWidth + gap));
-          rows = Math.ceil(model.grid.length / columns);
-          slotWidth = (width - (columns - 1) * gap) / columns;
-          slotHeight = (minHeight - (rows - 1) * gap) / rows;
-        }
-        if (slotHeight < slotMinHeight) slotHeight = slotMinHeight;
-        // Impose a minimum and maximum aspect ratio on the slots
-        const slotAspectRatio = slotWidth / slotHeight;
-        if (slotAspectRatio > slotMaxAspectRatio)
-          slotWidth = slotHeight * slotMaxAspectRatio;
-        else if (slotAspectRatio < slotMinAspectRatio)
-          slotHeight = slotWidth / slotMinAspectRatio;
-        // TODO: We might now be hitting the minimum height or width limit again
-
-        return [gap, slotWidth, slotHeight];
-      }, [width, minHeight, model.grid.length]);
-
-      const [generation] = useReactiveState<number>(
-        (prev) => (prev === undefined ? 0 : prev + 1),
-        [model.grid, width, minHeight],
-      );
-
-      return (
-        <div
-          ref={ref}
-          data-generation={generation}
-          className={styles.scrolling}
-          style={
-            {
-              width,
-              "--gap": `${gap}px`,
-              "--width": `${Math.floor(slotWidth)}px`,
-              "--height": `${Math.floor(slotHeight)}px`,
-            } as GridCSSProperties
-          }
-        >
-          {model.grid.map((tile) => (
-            <Slot className={styles.slot} tile={tile.id} />
-          ))}
-        </div>
+        ),
       );
-    }),
-  },
+      let rows = Math.ceil(model.grid.length / columns);
+
+      let slotWidth = (width - (columns - 1) * gap) / columns;
+      let slotHeight = (minHeight - (rows - 1) * gap) / rows;
+
+      // Impose a minimum width and height on the slots
+      if (slotWidth < slotMinWidth) {
+        // In this case we want the slot width to determine the number of columns,
+        // not the other way around. If we take the above equation for the slot
+        // width (w = (W - (c - 1) * g) / c) and solve for c, we get
+        // c = (W + g) / (w + g).
+        columns = Math.floor((width + gap) / (slotMinWidth + gap));
+        rows = Math.ceil(model.grid.length / columns);
+        slotWidth = (width - (columns - 1) * gap) / columns;
+        slotHeight = (minHeight - (rows - 1) * gap) / rows;
+      }
+      if (slotHeight < slotMinHeight) slotHeight = slotMinHeight;
+      // Impose a minimum and maximum aspect ratio on the slots
+      const slotAspectRatio = slotWidth / slotHeight;
+      if (slotAspectRatio > slotMaxAspectRatio)
+        slotWidth = slotHeight * slotMaxAspectRatio;
+      else if (slotAspectRatio < slotMinAspectRatio)
+        slotHeight = slotWidth / slotMinAspectRatio;
+      // TODO: We might now be hitting the minimum height or width limit again
+
```

Comment:
what do you think about moving all this computation into a simple descriptive funtion and keep theis method short?
e.g. somthing like `computeSlotHeight(`...

Reply:
I've done exactly that in https://github.com/element-hq/element-call/pull/2417 :D
For the sake of time I won't try to reorder the commit stack to move that change into this PR. Hope that's fine!

Reply:
oh definitely not!
Nice to have this updated in the final version.